lib/bundler/gem_helper.rb

```diff
@@ -92,8 +92,11 @@ def install_gem(built_gem_path = nil, local = false)
 
     def rubygem_push(path)
       if Pathname.new("~/.gem/credentials").expand_path.exist?
-        sh("gem push '#{path}'")
-        Bundler.ui.confirm "Pushed #{name} #{version} to rubygems.org."
+        gem_command = "gem push '#{path}'"
+        allowed_push_host = @gemspec.metadata["allowed_push_host"]
```

Comment:
isn't it possible that `metadata` isn't implemented, on a sufficiently old RubyGems?